frontend/src/store/connection/actions.ts

```diff
@@ -0,0 +1,18 @@
+import {
+  ConnectionAction,
+  BACKEND_HEARTBEAT,
+  BACKEND_CONNECTION_UP,
+  BACKEND_CONNECTION_DOWN,
+} from './types';
+
+export function lostBackendConnection(): ConnectionAction {
+  return { type: BACKEND_CONNECTION_DOWN };
+}
+
+export function establishedBackendConnection(): ConnectionAction {
+  return { type: BACKEND_CONNECTION_UP };
+}
+
+export function receivedBackendHeartbeat(): ConnectionAction {
+  return { type: BACKEND_HEARTBEAT, backendHeartbeat: new Date() };
```

Comment:
It looks like we no longer need to create a `new Date()` in this action creator.